### PR TITLE
Additional fields for Fleet create form

### PR DESCRIPTION
**Pull request by rawagner**

### CODE REVIEW

File: src/app/components/Fleet/CreateFleet/ConfigTemplateForm.tsx [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+
+const InlineConfigForm: React.FC<GitConfigFormProps> = ({ index }) => {
+  const { values, setFieldValue } = useFormikContext<FleetFormValues>();
+  const template = values.configTemplates[index] as InlineConfigTemplate;
+  return (
+    <Grid hasGutter>
+      <FormGroup label="Source name" isRequired>
+        <TextInput
+          aria-label="Source name"
+          value={values.name}
```
Comment (celdrake) [inline src/app/components/Fleet/CreateFleet/ConfigTemplateForm.tsx:103]: I think this line is what causes that the "source name" becomes uneditable and takes whatever value is set on the Fleet name.

File: src/app/components/form/LabelsField.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
+import { Label, LabelGroup } from '@patternfly/react-core';
+import * as React from 'react';
+
+type LabelsFieldProps = {
+  labels: { [key: string]: string };
+  setLabels: (newLabels: { [key: string]: string }) => void;
+};
+
+const LabelsField: React.FC<LabelsFieldProps> = ({ labels, setLabels }) => {
```
Comment (celdrake) [inline src/app/components/form/LabelsField.tsx:9]: I've noticed a couple of things we can improve. We can add TODO to fix later, as you prefer.

1. Are labels without a value valid? If they are, we currently display `label=undefined` so we should display just `label`
2. By always adding the new label with key=`newLabel`, the behaviour is a bit weird. If we click multiple times on "Add label", I'm getting it replaces the existing one, so in effect it looks as if the button is not working. I wanted to do something like "create 3 labels", edit their values later. Also, if you edit the value and not the key, then clicking on "add label" overwrites the label.

**Review by celdrake — CHANGES_REQUESTED** [review]
On top of the specific comments, we should prevent submission of the form if there are errors. ATM you can submit the form without filling a single field. For most settings it creates an empty fleet, with the inline config the API throws a 400 error (I'm guessing due to `inline=undefined`).

File: src/app/components/Fleet/CreateFleet/ConfigTemplateForm.tsx (hunk 1 above)
Comment (rawagner) [reply src/app/components/Fleet/CreateFleet/ConfigTemplateForm.tsx:103]: good catch!

**Comment by rawagner** [comment]
I will be adding Yup validations as a followup, so I didnt bother to disable submit if the form is not valid yet.
Let me fix the add label behavior.

File: src/app/components/form/LabelsField.tsx (hunk 2 above)
Comment (celdrake) [reply src/app/components/form/LabelsField.tsx:9]: Works great now!

**Review by celdrake — APPROVED** [review]

**Pull request merged**